arch/risc-v/src/esp32c3/esp32c3_i2c.c

```diff
@@ -1037,14 +1037,30 @@ static int esp32c3_i2c_transfer(struct i2c_master_s *dev,
 
       esp32c3_i2c_init_clock(priv, msgs[i].frequency);
 
-      /* Reset I2C trace logic */
+      if (msgs[i].flags & I2C_M_NOSTART)
+        {
+          esp32c3_i2c_traceevent(priv, I2CEVENT_SENDBYTE, priv->bytes,
+                                 getreg32(I2C_SR_REG(priv->id)));
+          esp32c3_i2c_senddata(priv);
 
-      esp32c3_i2c_tracereset(priv);
+          if (priv->bytes == msgs[i].length)
+            {
+              priv->i2cstate = I2CSTATE_STOP;
+              if (msgs[i].flags & I2C_M_NOSTOP)
+                priv->i2cstate = I2CSTATE_FINISH;
```

Comment:
Optionnal
```suggestion
              if ((msgs[i].flags & I2C_M_NOSTOP) != 0)
                {
                  priv->i2cstate = I2CSTATE_FINISH;
                }
```
adding `{}` not optional